### ISSUE TITLE
[Feature] Implement PyTorch-like Sequential Model Support

### ISSUE THREAD

**Issue by Cedricyu** [Cedricyu/tinyDL#7]
# 🧩 Feature Request: Modularize Linear Layers into a Layer Module

## Description

Currently, the code initializes each `Linear` layer manually and adds its parameters individually to the `SGD` optimizer:

```cpp
Linear layer1(input_dim, hidden_dim);
Linear layer2(hidden_dim, output_dim);
optimizer.add_param(layer1.d_weight, ...);
optimizer.add_param(layer2.d_weight, ...);
```

### Motivation

Refactoring Linear layers into a modular design allows us to:
Add layers dynamically (e.g., via vector or configuration)
Automatically register parameters to the optimizer
Support future extensions (e.g., ReLU, BatchNorm)
Simplify training loop and improve code readability
